### PR TITLE
Improve config key descriptions

### PR DESCRIPTION
**Pull request by aledsage**
See individual commits for details.

Note the last two commits make significant changes to config:
* remove the default for `launch.command` of `VanillaSoftwareProcess`.
* Rename some config keys (deprecating the old names) for `latch.*` and `skip.*`.

### CODE REVIEW

**Comment by aledsage** [comment]
retest this please

File: core/src/main/java/org/apache/brooklyn/core/entity/Attributes.java [hunk 1]
```diff
@@ -52,11 +52,15 @@
 
     // TODO these should switch to being TemplatedStringAttributeSensorAndConfigKey
     BasicAttributeSensorAndConfigKey<String> DOWNLOAD_URL = new BasicAttributeSensorAndConfigKey<String>(
-            String.class, "download.url", "URL pattern for downloading the installer (will substitute things like ${version} automatically)");
+            String.class, 
+            "download.url", 
+            "URL pattern for downloading the installer (will substitute things like ${version} automatically)");
```
Comment (tbouron) [inline core/src/main/java/org/apache/brooklyn/core/entity/Attributes.java]: Shouldn't it just say `uses FreeMarker templating format` to be consistent with the other description below?

File: core/src/main/java/org/apache/brooklyn/core/entity/Attributes.java [hunk 2]
```diff
@@ -52,11 +52,15 @@
...
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
     BasicAttributeSensorAndConfigKey<Map<String,String>> DOWNLOAD_ADDON_URLS = new BasicAttributeSensorAndConfigKey(
-            Map.class, "download.addon.urls", "URL patterns for downloading named add-ons (will substitute things like ${version} automatically)");
+            Map.class, 
+            "download.addon.urls", 
+            "URL patterns for downloading named add-ons (will substitute things like ${version} automatically)");
```
Comment (tbouron) [inline core/src/main/java/org/apache/brooklyn/core/entity/Attributes.java]: Shouldn't it just say `uses FreeMarker templating format` to be consistent with the other description below?

File: core/src/main/java/org/apache/brooklyn/core/entity/BrooklynConfigKeys.java [hunk 3]
```diff
@@ -180,23 +194,59 @@
...
+            .build();
+    public static final ConfigKey<Boolean> LAUNCH_LATCH = ConfigKeys.builder(Boolean.class)
+            .name("launch.latch")
+            .description("Latch for blocking luanch; if non-null will wait for this to resolve (normal use is with '$brooklyn:attributeWhenReady')")
+            .build();
 
     public static final ConfigKey<Duration> START_TIMEOUT = newConfigKey(
-            "start.timeout", "Time to wait for process and for SERVICE_UP before failing (in seconds, default 2m)", Duration.seconds(120));
+            "start.timeout", 
+            "Time to wait, after launching, for SERVICE_UP before failing (default to '2m')",
```
Comment (tbouron) [inline core/src/main/java/org/apache/brooklyn/core/entity/BrooklynConfigKeys.java]: Not sure the default needs to be written in the description. Would be hard to maintain and the CLI / UI have access to this info anyway.

File: core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabric.java [hunk 4]
```diff
@@ -50,13 +51,20 @@
                     + "for each given location).",
             true);
     
+    @CatalogConfig(label = "Member spec")
     @SetFromFlag("memberSpec")
     ConfigKey<EntitySpec<?>> MEMBER_SPEC = ConfigKeys.newConfigKey(
-            new TypeToken<EntitySpec<?>>() {}, "dynamiccfabric.memberspec", "entity spec for creating new cluster members", null);
+            new TypeToken<EntitySpec<?>>() {}, 
+            "dynamiccfabric.memberspec", 
+            "entity spec for creating new members (one per location)",
```
Comment (tbouron) [inline core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabric.java]: s/entity/Entity/

File: core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabric.java [hunk 5]
```diff
@@ -50,13 +51,20 @@
...
+            "dynamiccfabric.memberspec", 
+            "entity spec for creating new members (one per location)", 
+            null);
 
     @SetFromFlag("firstMemberSpec")
     ConfigKey<EntitySpec<?>> FIRST_MEMBER_SPEC = ConfigKeys.newConfigKey(
-            new TypeToken<EntitySpec<?>>() {}, "dynamiccfabric.firstmemberspec", "entity spec for creating new cluster members", null);
+            new TypeToken<EntitySpec<?>>() {}, 
+            "dynamiccfabric.firstmemberspec", 
+            "entity spec for the first member",
```
Comment (tbouron) [inline core/src/main/java/org/apache/brooklyn/entity/group/DynamicFabric.java]: s/entity/Entity/

**Comment by aledsage** [comment]
Thanks @tbouron - comments addressed.

@drigodwin I believe you were also looking at this? Ready to merge?

File: core/src/main/java/org/apache/brooklyn/enricher/stock/Aggregator.java [hunk 6]
```diff
@@ -65,18 +66,22 @@
...
     public static final ConfigKey<String> QUORUM_CHECK_TYPE = ConfigKeys.newStringConfigKey(
-            "quorum.check.type", "The requirement to be considered quorate -- possible values: " +
+            "quorum.check.type", 
+            "The requirement to be considered quorate (used with transformation of type 'isQuorate') -- possible values: " +
                     "'all', 'allAndAtLeastOne', 'atLeastOne', 'atLeastOneUnlessEmpty', 'alwaysHealthy'", "allAndAtLeastOne");
 
     public static final ConfigKey<Integer> QUORUM_TOTAL_SIZE = ConfigKeys.newIntegerConfigKey(
-            "quorum.total.size", "The total size to consider when determining if quorate", 1);
+            "quorum.total.size", 
+            "The total size to consider when determining if quorate (used iwth transformation of type 'isQuorate')", 1);
```
Comment (drigodwin) [inline core/src/main/java/org/apache/brooklyn/enricher/stock/Aggregator.java]: typo iwth -> with

File: core/src/main/java/org/apache/brooklyn/core/entity/BrooklynConfigKeys.java [hunk 7]
```diff
@@ -88,40 +93,48 @@
...
      * If this key is set on a {@link Location} then all entities in that location will be treated in this way, again as with {@link #ENTITY_STARTED}.
      *
-     * @see #ENTITY_STARTED
+     * @see #SKIP_ENTITY_START
      */
-    public static final ConfigKey<Boolean> SKIP_ENTITY_START_IF_RUNNING = newBooleanConfigKey("entity.running", "Skip the startup process entirely, if service already running");
+    public static final ConfigKey<Boolean> SKIP_ENTITY_START_IF_RUNNING = ConfigKeys.builder(Boolean.class)
+            .name("skip.start.ifRunning") 
+            .deprecatedNames("entity.running") 
+            .description("Whether to skip the startup process entirely, but only if it already running")
```
Comment (ahgittin) [inline core/src/main/java/org/apache/brooklyn/core/entity/BrooklynConfigKeys.java]: `Whether to skip the startup process if the entity is detected as already running` ?

File: core/src/main/java/org/apache/brooklyn/enricher/stock/Joiner.java [hunk 8]
```diff
@@ -47,31 +47,42 @@
...
+    
     public static final ConfigKey<Sensor<?>> SOURCE_SENSOR = ConfigKeys.newConfigKey(new TypeToken<Sensor<?>>() {},
-            "enricher.sourceSensor");
+            "enricher.sourceSensor",
+            "The sensor (expected to be of type Map or Iterable) whose change triggers re-evaluation of the target value");
+    
     public static final ConfigKey<Sensor<?>> TARGET_SENSOR = ConfigKeys.newConfigKey(new TypeToken<Sensor<?>>() {},
-            "enricher.targetSensor");
+            "enricher.targetSensor",
+            "The sensor that will be set on the associated entity, with the target value");
```
Comment (ahgittin) [inline core/src/main/java/org/apache/brooklyn/enricher/stock/Joiner.java]: `The sensor to be set on the associated entity with the value computed here` ?

**Comment by ahgittin** [comment]
a couple minor suggestions of wording

LGTM

as per @tbouron - icons v important (more important than config description IMHO!)

**Comment by tbouron** [comment]
@aledsage I just realised that you miss one more bom file in `karaf/init/src/main/resources/catalog-classes.bom`. This one is the base for the karaf distribution.

(I think we should reuse the same base bom files for the classic and Karaf distributions, but not in the scope of this PR)

**Comment by aledsage** [comment]
@ahgittin @tbouron @drigodwin thanks for comments - incorporated in an additional commit. Is it ok to merge (with icons being separate from this PR)?

**Comment by tbouron** [comment]
@aledsage Ok to me 👍  (build failure looks unrelated)

**Pull request merged**